components/sup/src/manager/self_updater.rs

```diff
@@ -15,52 +16,67 @@ use tokio::{self,
             time as tokiotime};
 
 pub const SUP_PKG_IDENT: &str = "core/hab-sup";
-const DEFAULT_PERIOD: Duration = Duration::from_secs(60);
-
-habitat_core::env_config_duration!(
-    /// Represents how far apart checks for updates are, in milliseconds.
-    SelfUpdatePeriod,
-    HAB_SUP_UPDATE_MS => from_millis,
-    DEFAULT_PERIOD);
 
 pub struct SelfUpdater {
     rx:             Receiver<PackageInstall>,
     current:        PackageIdent,
     update_url:     String,
     update_channel: ChannelIdent,
+    period:         DurationWithSplay,
+}
+
+/// The subset of data from `SelfUpdater` needed to spawn the updater task.
+struct Runner {
+    current:        PackageIdent,
+    update_url:     String,
+    update_channel: ChannelIdent,
+    period:         DurationWithSplay,
 }
 
-// TODO (CM): Want to use the Periodic trait here, but can't due to
-// how things are currently structured (The service updater had a worker)
+impl<T: Borrow<SelfUpdater>> From<T> for Runner {
+    fn from(other: T) -> Self {
+        let other = other.borrow();
+        Self { current:        other.current.clone(),
+               update_url:     other.update_url.clone(),
+               update_channel: other.update_channel.clone(),
+               period:         other.period, }
+    }
+}
 
 impl SelfUpdater {
-    pub fn new(current: &PackageIdent, update_url: String, update_channel: ChannelIdent) -> Self {
-        let rx = Self::init(current.clone(), update_url.clone(), update_channel.clone());
+    pub fn new(current: &PackageIdent,
+               update_url: String,
+               update_channel: ChannelIdent,
+               period: DurationWithSplay)
+               -> Self {
+        let runner = Runner { current: current.clone(),
+                              update_url: update_url.clone(),
+                              update_channel: update_channel.clone(),
+                              period };
+        let rx = Self::init(runner);
```

Comment:
I wonder if, now that we're exposing the auto-update-period tunable, we should incorporate a minimum allowed value of the current default 60s? Similar to how the [service_updater](https://github.com/habitat-sh/habitat/blob/119b59269008109138545eec32b22f1abe2e2a4c/components/sup/src/manager/service_updater/package_update_worker.rs#L28) has a minimum. Otherwise, it could potentially be somewhat misused and I don't think I've seen any demand for decreasing the value from 60.

Reply:
The use cases I have heard sited for very short periods is demos and testing. @ericcalabretta has sited those as motivation for not having a minimum. While I agree there is probably not a "real" use case for periods shorter than 60s, is it worth adding another flag that allows overriding the minimum?

Reply:
I understand the desire for a shorter period in testing and demos and therefore I retract my reservations. The additional override flag seems like just another variable to have to account for and document and won't really change any behaviors if a user is intent on having a lower interval across their fleet. We'll just have to adapt if this ends up impacting the req/s rate coming into the public Builder API.